### PR TITLE
feat: block beacon

### PR DESCRIPTION
**Pull request by robertsevernsentient**
config to block the beacon and method call to unblock beacon calls

RKT-8822

### CODE REVIEW

**Comment by robertsevernsentient** [comment]
hotfix-v1.8

File: src/beacon.js [hunk 1]
```diff
@@ -100,6 +101,11 @@ export default function Emitter(endpoint, context) {
     window.addEventListener('beforeunload', transmit);
   }
 
+  this.allowFlush = function() {
```
Comment (mattstrom) [inline src/beacon.js:104]: Choice of name. If my suggestion is wrong, then the purpose of the method isn't intuitive to me.
```suggestion
  this.unblockAndFlush = function() {
```

File: src/index.js [hunk 2]
```diff
@@ -43,8 +43,8 @@ function EvolvClient(options) {
 
   const store = options.store || new Store(options);
   const context = options.context || new Context(store);
-  const contextBeacon = options.analytics ? new Beacon(options.endpoint + '/' + options.environment + '/data', context) : null;
-  const eventBeacon = options.beacon || new Beacon(options.endpoint + '/' + options.environment + '/events', context);
+  const contextBeacon = options.analytics ? new Beacon(options.endpoint + '/' + options.environment + '/data', context, options.blockEvents) : null;
```
Comment (mattstrom) [inline src/index.js:46]: Block is not entirely precise based on my reading. Maybe `buffer`?
```suggestion
  const contextBeacon = options.analytics ? new Beacon(options.endpoint + '/' + options.environment + '/data', context, options.bufferEvents) : null;
```

File: src/tests/index.test.js [hunk 3]
```diff
@@ -719,6 +719,155 @@ describe('Evolv client integration tests', () => {
       expect(messages[18].sid).to.equal(sid)
     });
   });
+
+  describe('prevent beacon', () => {
+    it('should block beacoon calls', async () => {
```
Comment (mattstrom) [inline src/tests/index.test.js:724]: 🦝 
```suggestion
    it('should block beacon calls', async () => {
```

**Review by mattstrom — APPROVED** [review]

**Pull request merged**